apps/challenges/views.py

```diff
@@ -1045,95 +1045,148 @@ def download_all_submissions(request, challenge_pk, challenge_phase_pk, file_typ
+                                    submission['submission_result_file'],
+                                    submission['stdout_file'],
+                                    submission['stderr_file'],
+                                    submission['created_at'],
+                                     ])
+                return response
+            else:
+                response_data = {
+                    'error': 'You are neither host nor participant of the challenge!'}
+                return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            response_data = {'error': 'The file type requested is not valid!'}
+            return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+    elif request.method == 'POST':
         if is_user_a_host_of_challenge(user=request.user, challenge_pk=challenge_pk):
+            fieldsTable = {
+                'participant_team': 'Team Name',
+                'participant_team_members': 'Team Members',
+                'participant_team_members_email': 'Team Members Email Id',
+                'challenge_phase': 'Challenge Phase',
+                'status': 'Status',
+                'created_by': 'Created By',
+                'execution_time': 'Execution Time(sec.)',
+                'submission_number': 'Submission Number',
+                'input_file': 'Submitted File',
+                'stdout_file': 'Stdout File',
+                'stderr_file': 'Stderr File',
+                'created_at': 'Submitted At (mm/dd/yyyy hh:mm:ss)',
+                'submission_result_file': 'Submission Result File',
+                'submission_metadata_file': 'Submission Metadata File',
+            }
             submissions = Submission.objects.filter(
                 challenge_phase__challenge=challenge).order_by('-submitted_at')
             submissions = ChallengeSubmissionManagementSerializer(
                 submissions, many=True, context={'request': request})
             response = HttpResponse(content_type='text/csv')
             response['Content-Disposition'] = 'attachment; filename=all_submissions.csv'
             writer = csv.writer(response)
-            writer.writerow(['id',
-                             'Team Name',
-                             'Team Members',
-                             'Team Members Email Id',
-                             'Challenge Phase',
-                             'Status',
-                             'Created By',
-                             'Execution Time(sec.)',
-                             'Submission Number',
-                             'Submitted File',
-                             'Stdout File',
-                             'Stderr File',
-                             'Submitted At',
-                             'Submission Result File',
-                             'Submission Metadata File',
-                             ])
+            fields = [fieldsTable[field] for field in request.data]
```

Comment:
`[fieldsTable[field] for field in request.data]` -> `[fields_to_export[field] for field in request.data]`